project/preprocess_dockerfile.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+echo "#!/bin/bash"
+echo "cat << EOF"
+regex='^(if|elif|else|fi).*$'
+regex2='.*\\$'
+regex3='(.*)(\$|`)(.*)'
+while IFS='' read -r line
+do
+    if [[ $line =~ $regex ]]; then
+        echo "EOF"
+        echo "$line" 
+	echo "cat << EOF"
+    else
+	if [[ $line =~ $regex2 ]]; then
+		line="$line\\"
+	fi    
+	if [[ $line =~ $regex3 ]]; then
+        	templine="${BASH_REMATCH[1]}"
+		result="\\${BASH_REMATCH[2]}${BASH_REMATCH[3]}"
+		while [[ $templine =~ $regex3 ]]; do
+			templine="${BASH_REMATCH[1]}"
+			result="\\${BASH_REMATCH[2]}${BASH_REMATCH[3]}$result"
+		done
+		line="$templine$result"
+    	fi
+        echo "$line"
+    fi  
+done <$(dirname "$BASH_SOURCE")/../Dockerfile.in
+echo "EOF"
+
```

Comment:
This whole script is very complicated and fragile.  Why not just have a simple script that outputs a generated `Dockerfile` given some very simple knobs?  We're not trying to _create_ a template engine here -- we're trying to use shell scripting itself _as_ the template engine.